natlib/svcd.c

```diff
@@ -185,3 +186,86 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+//////////////////////////////////////////////////////////////////////////////
+// SVCD.notify() implementation
+// Maintainers: Sam Kumar <samkumar@berkeley.edu>
+//              Leonard Truong <leonardtruong@berkeley.edu>
+//              Michael Chen <mc.michaelchen.us@gmail.com>
+/////////////////////////////////////////////////////////////
+int notify_cord(lua_State* L) {
+    int header_index;
+    lua_pushvalue(L, lua_upvalueindex(1));
+    for (lua_pushnil(L); lua_next(L, 1); lua_pop(L, lua_gettop(L) - 2)) {
+	lua_pushlightfunction(L, arr_create);
+	lua_pushnumber(L, 1);
+	lua_pushnumber(L, ARR_TYPE_UINT16);
+	lua_call(L, 2, 1);
+	header_index = lua_gettop(L);
+	lua_pushstring(L, "set");
+	lua_gettable(L, -2);
+	lua_pushnumber(L, 1);
+	lua_pushvalue(L, 3);
+	lua_call(L, 2, 0);
+	lua_pushlightfunction(L, libstorm_net_sendto);
+	lua_getglobal(L, "SVCD");
+	lua_pushstring(L, "ncsock");
+	lua_gettable(L, -2);
+	lua_pushstring(L, "as_str");
```

Comment:
<code><pre>lua_pushnumber(L, ARR_TYPE_UINT16)</pre></code> is the correct code. In Lua you would do <code><pre>storm.array.create(1, storm.array.UINT16)</pre></code>. What you're doing in the Lua code is calling storm.array.create with two arguments: the length of the array, and a constant to say how big each element is. So we push storm_array_create (the function), 1 (the length), ARR_TYPE_UINT16 (the constant that indicates the length), and then invoke lua_call.

So to answer your question, storm.array.UINT16 is a constant that tells the create function how big each element of the array should be. In C you can access that constant as ARR_TYPE_UINT16. Make sense?